> Rather than manually keeping DAB's OIDC URL patterns in sync with DOT, consider adding a test that automatically detects which OIDC endpoints DAB has registered (by intersecting DAB's url names with DOT's oidc_urlpatterns) and asserts the regex patterns match. 

I added a test file locally with your suggestion, which ran perfectly. but there's two problems:

1. Our regex pattern for `jwks.json` escapes the `.` but the [d-o-t pattern does not](https://github.com/django-oauth/django-oauth-toolkit/blob/2.3.0/oauth2_provider/urls.py#L39), so the test will fail. I could adopt the "buggy" behavior to pass this test, but that seems like an antipattern. I could also remove this from the set for versions of d-o-t without the fix, maybe I'll do that.
2. The pattern for `userinfo/` doesn't match. I see you mentioned "This would have caught the `userinfo` mismatch" but I don't have context on that. I can change that (we don't have a specific requirement for the `userinfo` URL for this feature but the discovery view needs to be able to `reverse` it), but is something else expecting that?

```
================================================================== short test summary info ===================================================================
FAILED test_app/tests/oauth2_provider/test_oidc_urlpatterns.py::test_oidc_url_patterns_match_dot[jwks-info] - AssertionError: URL pattern for 'jwks-info' differs from django-oauth-toolkit. DAB: ^\.well-known/jwks\.json$, DOT: ^\.well-known/jwks.json$
assert '^\\.well-known/jwks\\.json$' == '^\\.well-known/jwks.json$'
  
  - ^\.well-known/jwks.json$
  + ^\.well-known/jwks\.json$
  ?                   +
FAILED test_app/tests/oauth2_provider/test_oidc_urlpatterns.py::test_oidc_url_patterns_match_dot[user-info] - AssertionError: URL pattern for 'user-info' differs from django-oauth-toolkit. DAB: ^userinfo$, DOT: ^userinfo/$
assert '^userinfo$' == '^userinfo/$'
  
  - ^userinfo/$
  ?          -
  + ^userinfo$
```

--- a/ansible_base/lib/testing/util.py
+++ b/ansible_base/lib/testing/util.py
@@ -77,12 +77,27 @@ class StaticResourceAPIClient(ResourceAPIClient):
 
 
 @contextmanager
-def feature_flag_enabled(flag_name):
-    """Context manager to temporarily enable a feature flag"""
+def _feature_flag_state(flag_name, desired_state):
+    """Context manager to temporarily set a feature flag to a desired state"""
     was_enabled = flag_state(flag_name)
-    enable_flag(flag_name)
+    if desired_state:
+        enable_flag(flag_name)
+    else:
+        disable_flag(flag_name)
     try:
         yield
     finally:
-        if not was_enabled:
+        if was_enabled:
+            enable_flag(flag_name)
+        else:
             disable_flag(flag_name)
+
+
+def feature_flag_enabled(flag_name):
+    """Context manager to temporarily enable a feature flag"""
+    return _feature_flag_state(flag_name, desired_state=True)
+
+
+def feature_flag_disabled(flag_name):
+    """Context manager to temporarily disable a feature flag"""
+    return _feature_flag_state(flag_name, desired_state=False)

--- a/test_app/tests/oauth2_provider/test_oidc_urlpatterns.py
+++ b/test_app/tests/oauth2_provider/test_oidc_urlpatterns.py
@@ -1,0 +1,19 @@
+import pytest
+
+from oauth2_provider.urls import oidc_urlpatterns as dot_oidc_urlpatterns
+from ansible_base.oauth2_provider.urls import oauth_urls
+
+dot_oidc_names = {p.name for p in dot_oidc_urlpatterns}
+dab_oidc_names = {
+    getattr(p, 'name', None) for p in oauth_urls
+} & dot_oidc_names
+
+@pytest.mark.parametrize("name", sorted(dab_oidc_names))
+def test_oidc_url_patterns_match_dot(name):
+    """Ensure our OIDC URL patterns stay aligned with the pinned DOT version."""
+    dot_pattern = next(p for p in dot_oidc_urlpatterns if p.name == name)
+    dab_pattern = next(p for p in oauth_urls if getattr(p, 'name', None) == name)
+    assert dab_pattern.pattern._regex == dot_pattern.pattern._regex, (
+        f"URL pattern for '{name}' differs from django-oauth-toolkit. "
+        f"DAB: {dab_pattern.pattern._regex}, DOT: {dot_pattern.pattern._regex}"
+    )

--- a/test_app/tests/oauth2_provider/views/test_openid_configuration.py
+++ b/test_app/tests/oauth2_provider/views/test_openid_configuration.py
@@ -14,7 +14,9 @@ def test_oauth2_provider_openid_configuration_valid_issuer_url(client):
     with feature_flag_enabled('FEATURE_OIDC_WORKLOAD_IDENTITY_ENABLED'):
         with override_settings(OAUTH2_PROVIDER={'OIDC_ENABLED': True}):
             url = get_relative_url("oauth2_provider:oidc-connect-discovery-info")
-            response_json = client.get(url).json()
+            response = client.get(url)
+            assert response.status_code == 200, f"Expected 200, got {response.status_code}"
+            response_json = response.json()
             assert response_json['issuer'].endswith(
                 '/o'
             ), "issuer in discovery metadata is expected to end with /o and match the authorization root view, otherwise discovery will fail!"

